pkgs/top-level/all-packages.nix

```
@@ -7137,8 +7137,8 @@ in
 
   all-cabal-hashes = callPackage ../data/misc/hackage { };
 
-  # Build with ghc 8.4 due to https://github.com/NixOS/nixpkgs/issues/53597
-  purescript = haskell.lib.justStaticExecutables haskell.packages.ghc844.purescript;
+  purescript = callPackage ../development/compilers/purescript/purescript { inherit pkgs; };
```

```suggestion
  purescript = callPackage ../development/compilers/purescript/purescript { };
```

will do the trick with the above suggested changes.

---

Is it necessary to use callPackage instead of simply importing the derivation and providing pkgs? Or is it that it's expected all derivations in nixpkgs should be used via callPackage?